File: code/client-side-impl/src/application/model/local_impl/profile/credentials/Credential.java
package application.model.local_impl.profile.credentials;

import java.io.Serializable;
import java.util.Objects;

/**
 * The Class Credential.
 * @author Jeffrey Gaines
 * @version Sprint 1
 */
public class Credential implements Serializable  {
	private static final long serialVersionUID = 1L;
	private String username;
	private String password;

	private static final String USERNAME_MUST_BE_VALID = "username inputted must not be null or empty";
	private static final String PASSWORD_MUST_BE_VALID = "password inputted must not be null or empty";

	/**
	 * Instantiates a new credential.
	 *
	 * @param username the username
	 * @param password the password
	 */
	public Credential(String username, String password) {
		if (username == null) {
			throw new NullPointerException(USERNAME_MUST_BE_VALID);
		} else if (username.isBlank()) {
			throw new IllegalArgumentException(USERNAME_MUST_BE_VALID);
		}
		if (password == null) {
			throw new NullPointerException(PASSWORD_MUST_BE_VALID);
		} else if (password.isBlank()) {
			throw new IllegalArgumentException(PASSWORD_MUST_BE_VALID);
		}
		this.username = username;
		this.password = password;
	}

	@Override
	public boolean equals(Object obj) {
		if (obj == this) {
			return true;
		}
		if (!(obj instanceof Credential)) {
			return false;
		}
		var credential = (Credential) obj;
		var objectEqual = credential.getUsername().toLowerCase().equals(this.username.toLowerCase());
		return objectEqual;
	}
	
	@Override
	public int hashCode() {
		int result = Objects.hash(this.username.toLowerCase());
		return result; 
	}

	/**
	 * Gets the username.
	 *
	 * @return the username
	 */
	public String getUsername() {
		return this.username;
	}

	/**
	 * Sets the username.
	 *
	 * @param username the new username
	 */
	public void setUsername(String username) {
		if (username == null) {
			throw new NullPointerException(USERNAME_MUST_BE_VALID);
		} else if (username.isBlank()) {
			throw new IllegalArgumentException(USERNAME_MUST_BE_VALID);
		}
		this.username = username;
	}

	/**
	 * Gets the password.
	 *
	 * @return the password
	 */
	public String getPassword() {
		return this.password;
	}

	/**
	 * Sets the password.
	 *
	 * @param password the new password
	 */
	public void setPassword(String password) {
		if (password == null) {
			throw new NullPointerException(PASSWORD_MUST_BE_VALID);
		} else if (password.isBlank()) {
			throw new IllegalArgumentException(PASSWORD_MUST_BE_VALID);
		}
		this.password = password;
	}
}

File: code/server-side-impl/src/src/model/profile/credentials/credential.py
'''
Created on Mar 10, 2024

@author: Jeffrey
'''
    
class Credential:
    """
    A class to represent user credentials.

    Attributes:
        username (str): The username of the user.
        password (str): The password of the user.
    """

    def __init__(self, username, password):
        """
        Initializes a Credential object with the given username and password.

        Args:
            username (str): The username for the credential.
            password (str): The password for the credential.

        Raises:
            Exception: If username or password is None.
        """
        if username is None:
            raise Exception("username is none")
        if password is None:
            raise Exception("password is none")
        self.username = username
        self.password = password

    def get_username(self):
        """
        Returns the username of the credential.

        Returns:
            str: The username of the credential.
        """
        return self.username
    
    def set_username(self, new_username):
        """
        Sets a new username for the credential.

        Args:
            new_username (str): The new username to be set.

        Raises:
            Exception: If new_username is None.
        """
        if new_username is None:
            raise Exception("username is none")
        self.username = new_username

    def get_password(self):
        """
        Returns the password of the credential.

        Returns:
            str: The password of the credential.
        """
        return self.password
    
    def set_password(self, new_password):
        """
        Sets a new password for the credential.

        Args:
            new_password (str): The new password to be set.

        Raises:
            Exception: If new_password is None.
        """
        if new_password is None:
            raise Exception("password is none")
        self.password = new_password
